pyuvdata/uvdata/uvdata.py

```diff
@@ -1346,106 +1424,132 @@ def _clear_unused_phase_centers(self):
-        for defunct_name in defunct_list:
-            self._remove_phase_center(defunct_name)
+        for defunct_id in defunct_list:
+            self._remove_phase_center(defunct_id)
 
-    def _check_for_unphased(self):
+    def _check_for_unprojected(self):
         """
-        Check which Nblts are unphased in a multi phase center dataset.
+        Check which Nblts are unprojected in a multi phase center dataset.
 
         This convenience method returns back a boolean mask to identify which data
-        along the Blt axis contains unphased objects (which is only applicable when
+        along the Blt axis contains unprojected objects (which is only applicable when
         multi_phase_center=True)
 
         Returns
         -------
         blt_mask : ndarray of bool
-            A boolean mask for identifying which elements contain unphased objects
+            A boolean mask for identifying which elements contain unprojected objects
         """
         if self.multi_phase_center:
-            # Check and see if we have any unphased objects, in which case
+            # Check and see if we have any unprojected objects, in which case
             # their w-values should be zeroed out.
-            nophase_dict = {
-                self.phase_center_catalog[name]["cat_id"]: self.phase_center_catalog[
-                    name
-                ]["cat_type"]
-                == "unphased"
-                for name in self.phase_center_catalog.keys()
-            }
+            nophase_list = [
+                cat_id
+                for cat_id, cat_dict in self.phase_center_catalog.items()
+                if cat_dict["cat_type"] == "unprojected"
+            ]
 
-            # Use dict to construct a bool array
-            blt_mask = np.array(
-                [nophase_dict[idx] for idx in self.phase_center_id_array], dtype=bool
-            )
+            # Construct a bool mask
+            blt_mask = np.isin(self.phase_center_id_array, nophase_list)
         else:
             # If not multi phase center, we just need to check the phase type
             blt_mask = np.repeat(self.phase_type == "drift", self.Nblts)
 
         return blt_mask
 
-    def rename_phase_center(self, old_name, new_name):
+    def rename_phase_center(self, old_name=None, new_name=None, cat_id=None):
         """
         Rename a phase center/catalog entry within a multi phase center data set.
 
         Parameters
         ----------
         old_name : str
-            Phase center name for the data to be renamed.
+            Optional argument, phase center name for the target to be renamed. Note that
+            either `old_name` or `cat_id` must be specified.
         new_name : str
             New name for the phase center.
+        cat_id : int or list of int
+            Optional argument, phase center IDs number for the target to be renamed.
+            Note that either `old_name` or `cat_id` must be specified.
+
```

Comment:
Yeah, I think I like this approach better -- I put it in its current form trying to not break the existing argument order, but your suggestion above would seemingly preserve that. Thanks!

Reply:
Okay I like it. I think my only gripe now is we check `if isinstance(catalog_identifer, str)` and do the string part, but then just assume it must be an Int otherwise. Might be worth having a block at the top to error if it is not a str or an int?

Reply:
I can add that easily.